uk.co.compucorp.civicrm.hrleaveandabsences/xml/schema/CRM/HRLeaveAndAbsences/LeaveBalanceChange.xml

```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
+
+<table>
+  <base>CRM/HRLeaveAndAbsences</base>
+  <class>LeaveBalanceChange</class>
+  <name>civicrm_hrleaveandabsences_leave_balance_change</name>
+  <comment>Store balance changes to a Leave Period Entitlement</comment>
+  <add>4.4</add>
+  <log>true</log>
+
+  <field>
+    <name>id</name>
+    <type>int unsigned</type>
+    <required>true</required>
```

Comment:
Yes, I primary key cannot be null, but in MySQL (don't know about other databases), this statement is valid:
```SQL
CREATE TABLE `table` (
  `id` INT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`));
```
With the <required>true</true> a NOT NULL will be added to the statement. In the end both will work the same way (MySQL automatically adds the not null the the primary key, if it's not present), but I prefer to keep the required.
Also, you're almost right about the BAO and the create method. The required info won't have any effect in that method, but if you're in a form it will be used as a validation rule.